### ISSUE TITLE
when I use the unet to predict, I get some problem.

### ISSUE THREAD

**Issue by Lone5267** [booz-allen-hamilton/DSB3Tutorial#13]
when i use the 'unet.hdf5' that you given, i get this
![image](https://cloud.githubusercontent.com/assets/10320669/23250891/ed87bbb6-f9e5-11e6-9ff5-fa860ce39c21.png)

then I try the one i trained, i get  this
![image](https://cloud.githubusercontent.com/assets/10320669/23251052/b7cb563a-f9e6-11e6-9563-84bd637d0e9a.png)

Is that overfitting? I change the nb_epoch to 40.
```
model.fit(imgs_train, imgs_mask_train, batch_size=2, nb_epoch=40, verbose=1, shuffle=True,
              callbacks=[model_checkpoint])
```
May you share the parameters that you train on TitanX's ? tanks...

**Comment by dearkafka** [comment]
same result for some slices, but surprisingly some are good

**Comment by bharatv007** [comment]
I am getting similar results, I trained it from scratch, ran 50 epochs. It will be good to know if someone has solved it.

**Comment by dearkafka** [comment]
cropping image & masks into patches and training them in unet significantly improve everything as good pre-processing does. got 0.6 dice on validation and its pretty much ok

**Comment by vuongdanghuy** [comment]
can you explain what is your "patches"? did you resize the image and mask into different size and train it with UNET or anything? right now i'm training my UNET with image in size 128x128.

**Comment by dearkafka** [comment]
No, I do crops around nodules, some fixed-size parallelepiped. You do not actually need full image for segmentation

**Comment by bharatv007** [comment]
I had normalised the data set(just the data and mask), and gave it to unet, no pre-processing, getting good results after 100 epochs.

**Comment by jiandai** [comment]
Same issue when applying the pretrained "unet.hdf5" provided by this repos. I know this may be too much to ask but would you mind to provide a better/improved version of pretained weights? Thx!
![figure_1](https://cloud.githubusercontent.com/assets/5249054/24509220/ab5e7248-151a-11e7-9475-ce6eba3b300a.png)

**Comment by dszhengyu** [comment]
facing same problem
try to do some crops around nodule as @dearkafka said

**Comment by jt827859032** [comment]
I've trained my data with unet the tutorial code provided ,same situation happened that all the pixel values of the predict mask were barely zeros. Anybody can figure out what's wrong ?

**Comment by prateeknepaliya09** [comment]
@Lone5267  @jiandai  @dearkafka Can you share how did you use unet.hdf5. I'm kind of stuck on how to use this for a particular prediction ?

**Comment by dearkafka** [comment]
@prateeknepaliya09 Let me be precisely clear, I do not use this particular unet for prediction, because, as stated above it's not clear what the normalization of image should actually be, beacuse the stated one makes strange blurry elipses insteas of small nodule regions. However, it is actually relatively easy to train unet from scratch and get dice in 0.4-0.6 or higher, up to 0.8. If you face problems with training some unet architecture for this dataset I would suggest inset BatchNorm layers almost everywhere in conv blocks. Wihout them I could not get any progress. Or, if you would not like BatchNorm, have a look at selu activation which also helps

**Comment by orangeYao** [comment]
Exactly same results as you guys.

**Comment by yyyamada** [comment]
My results are exactly the same as yours.
Mean Dice Coeff : 0.332198897668
Loss: loss : 0.1753

**Comment by DyerLee** [comment]
> My results are exactly the same as yours.
> Mean Dice Coeff : 0.332198897668
> Loss: loss : 0.1753
Excuse me, how is your code executed? Thank you
